modules/gapi/src/api/kernels_video.cpp

```diff
@@ -57,5 +57,10 @@ GMat BackgroundSubtractor(const GMat& src, const BackgroundSubtractorParams& bsp
     return GBackgroundSubtractor::on(src, bsp);
 }
 
+GMat KalmanFilter(const GMat& m, const cv::GOpaque<bool>& have_m, const GMat& c, const KalmanParams& kp)
```

Comment:
So, again, why do we need this flag?
What data is passed in as `m` if this flag is `false`? I mean, in runtime.

Reply:
We need this flag because measurement matrix is optional for each particular iteration of algorithm. Kalman if measurement matrix is given (haveMeasurements == true) corrected state will be returned  which corresponds to the pipeline cv::KalmanFilter::predict(control) -> cv::KalmanFilter::correct(measurement). Otherwise, predicted state will be returned which corresponds to the call of cv::KalmanFilter::predict(control) only.

Reply:
I think, user will have to pass any matrix with appropriate sizes/type, it will not be used.
This is required for the problem: "we have the measurement with all zeros" and "we don't have a measurement on this iteration" are different cases.

Reply:
Orest, all your design questions and proposals you should raised at design meetings. There were several meetings and you had ample opportunity to speak. Also mail thread for Kalman design discussion was organized more than week ago. Why didn't you submit your design proposal on time? Design was created, discussed and approved. So, now you shouldn't review my snippet for design mistakes. And I shouldn't discuss design with you now. And it shouldn't defend this design in front of you. You break our workflow.
If you have any questions to this design, please organize one more meeting, invite all design members and we will discuss.

Reply:
No, no, this wasn't a question. Actually it is the explanation for the Dmitry's question

Reply:
Please avoid using such wording in a public space. Discuss with your manager.

Use clear polite answers like that `approved on design meeting as stated in the meeting minutes from <date>` if you don't want to provide normal comments.

> You break our workflow.

Review is part of workflow. We shouldn't avoid that.
Also you can't forbid to ask questions / provide comments or proposals. Even design issues.